crates/ide_assists/src/handlers/early_return.rs

```diff
@@ -56,7 +56,11 @@ pub(crate) fn convert_to_guarded_return(acc: &mut Assists, ctx: &AssistContext)
             match path.qualifier() {
                 None => {
                     let bound_ident = pat.fields().next().unwrap();
-                    Some((path, bound_ident))
+                    if ast::IdentPat::can_cast(bound_ident.syntax().kind()) {
```

Comment:
```suggestion
                    if ast::Pat::IdentPat(bound_ident) = bound_ident {
```
No need for casting sinec `Pat` is just a wrapper enum.

Reply:
I don't understand what you mean?

Reply:
ast::Pat::IdentPat(bound_ident) does not accept Pat, only `syntax::ast::IdentPat`

Reply:
I mean to extract the `IdentPat` out of it the pattern here but that doesn't really simplify anything actually so this is fine.